connection.go

```diff
@@ -152,20 +152,19 @@ func (ct *ConnectionTransportTLS) Dial(ctx context.Context) (
 	var conn net.Conn
 	err := runUnlessCanceled(ctx, func() error {
 		// load CA certificate
-		certs := x509.NewCertPool()
-		if !certs.AppendCertsFromPEM(ct.rootCerts) {
-			return errors.New("Unable to load root certificates")
+		config := ct.tlsConfig
+		if config == nil {
```

Comment:
`NewTLSConnectionWithTLSConfig` doesn't take a `rootCerts` argument, so `ct.rootCerts` will always be `nil` if that constructor is used. So if `nil` is passed into `NewTLSConnectionWithTLSConfig` for `config`, then a config with empty `rootCerts` will be used. Maybe I'm misreading it?

Looks like only `NewTLSConnectionWithProtocols` takes a `rootCerts` argument.